CMakeLists.txt

```diff
@@ -1,74 +1,155 @@
 # (by default, static build is off since libigl is a header-only library)
-option(LIBIGL_USE_STATIC_LIBRARY "Use libigl as static library" ON)
-
-# All dependencies that are downloaded as cmake projects and tested on the auto-builds are ON
-# (by default, all build options are off)
-option(LIBIGL_WITH_COMISO            "Use CoMiso"                   ON)
-option(LIBIGL_WITH_EMBREE            "Use Embree"                   ON)
-option(LIBIGL_WITH_OPENGL            "Use OpenGL"                   ON)
-option(LIBIGL_WITH_OPENGL_GLFW       "Use GLFW"                     ON)
-option(LIBIGL_WITH_OPENGL_GLFW_IMGUI "Use ImGui"                    ON)
-option(LIBIGL_WITH_PNG               "Use PNG"                      ON)
-option(LIBIGL_WITH_TETGEN            "Use Tetgen"                   ON)
-option(LIBIGL_WITH_TRIANGLE          "Use Triangle"                 ON)
-option(LIBIGL_WITH_PREDICATES        "Use exact predicates"         ON)
-option(LIBIGL_WITH_XML               "Use XML"                      ON)
-option(LIBIGL_WITH_PYTHON            "Use Python"                   OFF)
-### End
-
-if(${LIBIGL_WITH_PYTHON})
-	message(FATAL_ERROR "Python binding are in the process of being redone. Please use the master branch or refer to https://github.com/geometryprocessing/libigl-python-bindings for the developement version or https://anaconda.org/conda-forge/igl for the stable version.")
-endif()
+option(LIBIGL_USE_STATIC_LIBRARY "Use libigl as static library" ${LIBIGL_TOPLEVEL_PROJECT})
+
+# Main modules
+option(LIBIGL_WITH_EMBREE           "Build target igl::embree"           ${LIBIGL_TOPLEVEL_PROJECT})
+option(LIBIGL_WITH_GLFW             "Build target igl::glfw"             ${LIBIGL_TOPLEVEL_PROJECT})
+option(LIBIGL_WITH_IMGUI            "Build target igl::imgui"            ${LIBIGL_TOPLEVEL_PROJECT})
+option(LIBIGL_WITH_OPENGL           "Build target igl::opengl"           ${LIBIGL_TOPLEVEL_PROJECT})
+option(LIBIGL_WITH_PNG              "Build target igl::png"              ${LIBIGL_TOPLEVEL_PROJECT})
+option(LIBIGL_WITH_PREDICATES       "Build target igl::predicates"       ${LIBIGL_TOPLEVEL_PROJECT})
+option(LIBIGL_WITH_XML              "Build target igl::xml"              ${LIBIGL_TOPLEVEL_PROJECT})
+
+# Copyleft modules
+option(LIBIGL_COPYLEFT_WITH_CORE    "Build target igl_copyleft::core"    ${LIBIGL_TOPLEVEL_PROJECT})
+option(LIBIGL_COPYLEFT_WITH_CGAL    "Build target igl_copyleft::cgal"    ${LIBIGL_DEFAULT_CGAL_CORK})
+option(LIBIGL_COPYLEFT_WITH_COMISO  "Build target igl_copyleft::comiso"  ${LIBIGL_TOPLEVEL_PROJECT})
+option(LIBIGL_COPYLEFT_WITH_CORK    "Build target igl_copyleft::cork"    ${LIBIGL_DEFAULT_CGAL_CORK})
+option(LIBIGL_COPYLEFT_WITH_TETGEN  "Build target igl_copyleft::tetgen"  ${LIBIGL_TOPLEVEL_PROJECT})
 
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+# Nonfree modules
```

Comment:
Hmm so technically it seems that we have the following categories:
- Permissive: MIT-like license that allow commercial usage (igl::core, etc.)
- Copyleft: CGAL and other modules under GPL
- Noncommercial: Mosek and Matlab
- Proprietary: Triangle. Traditional copyright applies, and it is not clear how derivative work should be licensed (or even if we are allowed to produce derivative work by fixing bugs). Note that triangle has a proprietary license, even if the code itself is available online.

I think we should clearly indicate when a module is nonpermissive, such as copyleft or noncommercial/proprietary work. After all, all of libigl is available for academic usage, but I think it pays to be more transparent for non permissive works (like we do for copyleft).

Restricted might be a good alternative to encompass both noncommercial/proprietary licenses.

Reply:
I agree the names should indicate these things. I'm happy with restricted_use unless we can thing of something better.

I wish we had a better way to indicate with these that it's the _dependency_  that restricts the use, not the libigl code. For example, with matlab, it should be hypothetically possible to link to octave (free alternative; I've never tried this and have no clue if it works).

Reply:
I'd prefer `restricted` rather than `restricted_use` though, as it's just one word and less verbose to type/setup. Would that be acceptable?

For Octave bindings it would be interesting to support that, as we could add that to our Github Actions integration and make sure it works for everyone. (But I have very little interest in doing that myself though).

Reply:
I'm OK with `restricted` as long as we clearly spell out in the cmake comments and the documentation what is restricted and how (basically your categories above).

Reply:
Sure, I'll update that shortly.